lib/fog/vsphere/requests/compute/list_vm_interfaces.rb

```diff
@@ -29,31 +29,51 @@ class Real
   #unitNumber: 7,
   #
         def list_vm_interfaces(vm_id, datacenter = nil)
-          get_vm_ref(vm_id, datacenter).config.hardware.device.grep(RbVmomi::VIM::VirtualEthernetCard).map do |nic|
-            {
-              :name    => nic.deviceInfo.label,
-              :mac     => nic.macAddress,
-              :network => nic.backing.respond_to?("network") ? nic.backing.network.name : nic.backing.port.portgroupKey,
-              :status  => nic.connectable.status,
-              :summary => nic.deviceInfo.summary,
-              :type    => nic.class,
-              :key     => nic.key,
-            }
-          end
+          get_raw_interfaces(vm_id, datacenter).map {|nic| raw_to_hash nic}
         end
 
         def get_vm_interface(vm_id, options={})
+          if raw = get_raw_interface(vm_id, options)
```

Comment:
Is this supposed to be an assignment? I usually prefer not to make assignments inside of a conditional's predicate.